kafka-rest/src/main/java/io/confluent/kafkarest/resources/v2/PartitionsResource.java

```diff
@@ -192,17 +193,18 @@ public void onCompletion(
                 "Completed topic produce request id={} response={}",
                 asyncResponse, response
             );
-            asyncResponse.resume(response);
+            Response.Status requestStatus = Utils.produceRequestStatus(response);
+            asyncResponse.resume(Response.status(requestStatus).entity(response).build());
```

Comment:
I couldn't compile the project for some reason but I wanted to verify that `entity(response)` doesn't override the status set by `status(requestStatus)`